src/App.tsx

```diff
@@ -8,37 +8,24 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import {
+  fetchPosts,
+  removeSelectedPost,
+} from './components/postsSlice/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const author = useAppSelector(state => state.users.author);
+  const { items, selectedPost, loaded, hasError } = useAppSelector(
+    state => state.posts,
+  );
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(removeSelectedPost());
 
     if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
+      dispatch(fetchPosts(author.id));
     }
   }, [author]);
```

Comment:
disable eslint rule for this line

Reply:
Max Shepel told me not to disable rules because there can be situations when it can have consequences) 
But I disabled it as you asked)

Reply:
Shure, you need disabled cases only with specific situations. 
Most often it's use to effect deeps. Others, you need to fix with code ( not with disabling). 
Thanks)